src/frontend/Typechecker.ml

```diff
@@ -394,7 +394,40 @@ let check_fn ~is_cond_dist loc tenv id es =
               (Utils.normalized_name id.name)) ->
       Semantic_error.returning_fn_expected_nonfn_found loc id.name |> error
   | [] ->
-      Semantic_error.returning_fn_expected_undeclaredident_found loc id.name
+      ( match Utils.split_distribution_suffix id.name with
+      | Some (prefix, suffix) -> (
+          let known_families =
+            List.map
+              ~f:(fun (_, y, _, _) -> y)
+              Stan_math_signatures.distributions
+          in
+          let is_known_family s =
+            List.mem known_families s ~equal:String.equal
+          in
+          match suffix with
+          | ("lpmf" | "lumpf")
+            when Utils.is_distribution_name (prefix ^ "_lpdf") ->
+              Semantic_error.returning_fn_expected_wrong_dist_suffix_found loc
+                (prefix, suffix)
+          | ("lpdf" | "lumdf")
+            when Utils.is_distribution_name (prefix ^ "_lpmf") ->
```

Comment:
This conditional is too weak. See this test model (should add to tests):

completely_undefined:
```stan
model {
  target += foo_lpdf(1);
}
```
Current test output:
```
Semantic error in 'test.stan', line 2, column 12 to column 23:
   -------------------------------------------------
     1:  model {
     2:    target += foo_lpdf(1);
                     ^
     3:  }
   -------------------------------------------------

Function 'foo_lpdf' is not implemented for distribution 'foo', use 'foo_lpmf' instead.
```

I think `Utils.is_distribution_name` is checking if that is a well-formed name, _not_ checking that the function is defined. For that, the simplest check is to use `Env.mem`. That will still get fooled by something like

```stan
data {
   int foo_lpmf;
}
model {
  target += foo_lpdf(1);
}
```

but IMO this is probably okay, the user is being intentionally obtuse here.

Reply:
Added